src/sentry/static/sentry/app/components/noProjectMessage.jsx

```diff
@@ -1,31 +1,44 @@
 import React from 'react';
 import {Flex} from 'grid-emotion';
 import styled from 'react-emotion';
+import PropTypes from 'prop-types';
 
 import {t} from 'app/locale';
 import Button from 'app/components/button';
+import PageHeading from 'app/components/pageHeading';
 import Tooltip from 'app/components/tooltip';
 import SentryTypes from 'app/sentryTypes';
-import img from '../../../images/dashboard/hair-on-fire.svg';
+import space from 'app/styles/space';
+import img from '../../images/confused-io.png';
 
 export default class EmptyState extends React.Component {
   static propTypes = {
+    /* if the user has access to any projects, we show whatever
+    children are included. Otherwise we show the message */
+    children: PropTypes.node,
```

Comment:
I started it that way, but I had to basically copy pasta the "no projects" logic over and over again. I'm open to suggestions!

Reply:
Would it help if it was named `CheckForProject` to better indicate what the component is doing?

Reply:
Ideally I think we should have pass around an organization object with helper methods on it. I personally find duplication easier to read than this but ymmv

Reply:
I agree it is a little odd...and I would be fine with it if we can add a method to the `organization` object. @lynnagara can you show me an example somewhere in the codebase of how I might do that?

I should note that moving in this direction will be clearer but might not look any _cleaner_, since we will need to wrap large swaths of content in ternaries.

Reply:
I like the component having that logic fwiw

Reply:
since we seem to be split on this, and the work is already done, I'm gonna keep it the way it is.